enhancements/update/configurable-clusterversion-signature-stores.md

```diff
@@ -0,0 +1,188 @@
+To increase flexibility and improve disconnected cluster user experience, we need to provide a way for the CVO to 
+get the required signatures from custom signature stores or the OSUS running in the disconnected environment.
+
+
+## Motivation
+
+Today, in the disconnected environments, a user has to clone the signatures config map as well as the release images 
+then apply the configMap to each cluster for the update to proceed successfully.
+With [serve image signatures for disconnected environments], OSUS will be able to serve the cloned image signatures locally.
+
+Currently, there is no way for the CVO to fetch the signatures through custom URIs. This enhancement defines an API change in the CVO
+to accept a custom update-service URI which can be used to verify release signatures. 
+
+
+### User Stories
+
+* As a cluster administrator, I want the cluster's CVO to add new signature store URIs.
+* As a cluster administrator, I want the cluster's CVO to replace the existing signature store URIs with a user provided URI
+
+### Goals
+
+* To verify the release image signature in disconnected environments using OSUS signatures URI.
+* To replace existing signature stores with custom user-provided stores.
+
+### Non-Goals
+
+* This enhancement does not cover how OSUS running in the disconnected environment gets the signatures and if the 
+signatures are validated during the mirroring process.
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster admin will have to add the verification URI to CVO
+2. CVO will reach the newly added URI to verify the release images.
+
+### API Extensions
+
+* CVO will grow a new field, an array, which will be populated with the image verification URI.
+* This will be an array as it'll allow the admin to use multiple locally configured stores. 
+* Existing default signature stores will be used as fallback in-case custom stores don't contain signatures or are offline. 
+* If this property is empty, default stores will be used, similar to how we handle upstream for update service today.
+* Admins will still be able to use the existing config maps for signature verification. 
+* CVO will check the release signatures in the local ConfigMap first and will reach in parallel to other signature stores.
+
+This enhancement proposes to modify the [api/config/v1/types_cluster_version.go](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go) as following,
+
+A new field `SignatureStores` will be added to the [`ClusterVersionSpec`](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go#L40-L96) structure:
+
+```go
+type ClusterVersionSpec struct {
+	...
+	// SignatureStores contains the upstream URIs to verify release signatures.
+	// By default, CVO will use existing signature stores if this property is empty.
+	//
+	// Valid values are: empty array OR array of upstream signature store string.
```

Comment:
We typically talk about the values in the list, rather than the list itself.
Something along the lines of
```
// Items in this list should be a valid absolute URl of an upstream signature store as per <some http spec>.
```
Are there any restrictions on the URL, eg, can it contain ports, can it have a path, must it be https instead of http?